src/main/java/com/google/api/codegen/transformer/nodejs/NodeJSApiMethodParamTransformer.java

```diff
@@ -14,24 +14,165 @@
  */
 package com.google.api.codegen.transformer.nodejs;
 
+import com.google.api.codegen.config.MethodConfig;
 import com.google.api.codegen.transformer.ApiMethodParamTransformer;
 import com.google.api.codegen.transformer.MethodTransformerContext;
+import com.google.api.codegen.transformer.SurfaceNamer;
+import com.google.api.codegen.util.Name;
 import com.google.api.codegen.viewmodel.DynamicLangDefaultableParamView;
 import com.google.api.codegen.viewmodel.ParamDocView;
+import com.google.api.codegen.viewmodel.SimpleParamDocView;
+import com.google.api.tools.framework.model.Field;
+import com.google.api.tools.framework.model.TypeRef;
 import com.google.common.collect.ImmutableList;
 import java.util.List;
 
 public class NodeJSApiMethodParamTransformer implements ApiMethodParamTransformer {
   @Override
   public List<DynamicLangDefaultableParamView> generateMethodParams(
       MethodTransformerContext context) {
-    // TODO(eoogbe): implement this method when migrating to MVVM
-    return ImmutableList.<DynamicLangDefaultableParamView>of();
+    ImmutableList.Builder<DynamicLangDefaultableParamView> methodParams = ImmutableList.builder();
+    methodParams.addAll(generateDefaultableParams(context));
+
+    TypeRef arrayType = TypeRef.fromPrimitiveName("string").makeRepeated();
+
+    DynamicLangDefaultableParamView.Builder optionalArgs =
+        DynamicLangDefaultableParamView.newBuilder();
+    optionalArgs.name(context.getNamer().localVarName(Name.from("optional", "args")));
+    optionalArgs.defaultValue(context.getTypeTable().getZeroValueAndSaveNicknameFor(arrayType));
+    methodParams.add(optionalArgs.build());
+
+    return methodParams.build();
+  }
+
+  private List<DynamicLangDefaultableParamView> generateDefaultableParams(
+      MethodTransformerContext context) {
+    if (context.getMethod().getRequestStreaming()) {
+      return ImmutableList.of();
+    }
+    ImmutableList.Builder<DynamicLangDefaultableParamView> methodParams = ImmutableList.builder();
+    for (Field field : context.getMethodConfig().getRequiredFields()) {
```

Comment:
note: this could be rewritten as `Iterables.transform` -- not sure how worthwhile to do so. It's gonna be
```java
return Iterables.transform(context.getMethodConfig().getRequiredFields(), new Function() {
  @Override
  DynamicLangDefaultableParamView apply(Field field) {
    return DynamicDefaultableParamView.newBuilder().name(...).defaultValue("").build();
  }
});
```

(see http://google.github.io/guava/releases/snapshot/api/docs/com/google/common/collect/Iterables.html#transform-java.lang.Iterable-com.google.common.base.Function-)